packages/language-server/src/prisma-fmt/textDocumentCompletion.ts

```diff
@@ -0,0 +1,17 @@
+import prismaFmt from '@prisma/prisma-fmt-wasm'
+import * as lsp from 'vscode-languageserver'
+
+// This can't panic / throw exceptions. Any panic here should be considered a bug to be fixed.
+export default function textDocumentCompletion(
+  schema: string,
+  params: lsp.CompletionParams,
+): lsp.CompletionList {
+  // lsp.CompletionParams.textDocument doesn't match the LSP JSON-RPC protocol
```

Comment:
missing `uri` only? maybe add to comment

also: why?

Reply:
So the text document in a valid LSP request is `type TextDocument = { uri: string };`, basically. What the `params` argument is here has `type TextDocument = SomeSortOfTextDocumentObject`, where SomeSortOfTextDocumentObject has a bunch of methods, and `__uri` and `__content` properties. It's an object with logic, not the JSON protocol. Luckily the rest of the CompletionParams object appears to match the JSON protocol. I'll document that a bit more in the comments.